lib/geordi/hint.rb

```diff
@@ -0,0 +1,46 @@
+# Use the methods in this file to hint at other geordi features
+require File.expand_path('settings', __dir__)
+
+module Geordi
+  class Hint
+    class << self
+
+      def did_you_know(hints)
+        settings_probability =  Settings.new.did_you_know
+        should_print_hint = Random.new.rand(100) <= (settings_probability || 10)
+
+        generated_hints = hints.map { |hint| generate(hint) }.compact
```

Comment:
This `.compact` is not needed any more. And _if you like_, you can simplify the code with:
```suggestion
        generated_hints = hints.map(&method(:generate))
```

Reply:
Please remember to drop the `.compact`.